allow non ascii characters in swagger yaml specs

TNGPS opened the pull request:
the current version cannot handle non-ascii characters readling the yaml file.

assuming the file is encoded in UTF8 this will fix that.

traxair commented on tests/swagger.yaml:
```diff
@@ -6,6 +6,9 @@ info:
 
     For this sample, you can use the api key `special-key` to test the authorization filters
+    
+    NOTE: Swagger files mey contain non-acsii characters like µ, ü or ç and ñ, please save your file in UT8 to preserv them.
```
Replace with `NOTE: Swagger files may contain non-ascii characters like µ, ü or ç and ñ, please save your file in UTF-8 to preserve them.` please

traxair commented on swagger_parser/swagger_parser.py:
```diff
@@ -6,6 +6,7 @@
 import re
 import six
 import yaml
+import codecs
```
Could you keep an alphabetic ordering please? Put `import codec` right before `import datetime`

traxair commented:
Hey there, thanks for contributing! Could you correct the two minor items I've pointed at? Then it will be ready to merge.

traxair commented:
Could you also add some non-ascii characters in the swagger file we use for our tests? Thanks!

TNGPS commented:
Hello,
I corrected the two items you pointed out.
I assume the file `tests/swagger.yaml` is the one you use in testing the module ? This is why I added the non-ascii characters here in the info block.

traxair commented:
Oh right, I thought you had modified the README file. Could you also add a non-ascii characters in the description of some path? Like maybe here: https://github.com/TNGPS/swagger-parser/blob/patch-1/tests/swagger.yaml#L38
I'm not sure if we parse the global description of the swagger file, @cyprien-g ?

cyprieng commented:
@traxair it reads everything in the yaml file. So as the test do not raise any error, it means that utf8 is supported. But I am not against checking carefully that the utf8 string is correctly read.

traxair commented:
@TNGPS Feel free to add a bunch of utf-8 characters here and there then ;)

traxair commented:
Waiting for your addition and then we'll issue a new release ;)

traxair commented:
You need to add the new description in https://github.com/TNGPS/swagger-parser/blob/patch-1/tests/conftest.py#L45 and / or https://github.com/TNGPS/swagger-parser/blob/patch-1/tests/conftest.py#L55

flavianh commented:
@TNGPS If you want us to provide the functionality, you'll have to fix the tests ;). Come on you're nearly there!

TNGPS commented:
I tried to fix the tests, however it looks like not successfully :-(

any idea why the utf8 fixtures do not convert to unicode for the assertion ?

> UnicodeWarning: Unicode equal comparison failed to convert both arguments to Unicode - interpreting them as being unequal

cyprieng commented:
I think you should convert string in `conftest.py` to unicode: `u'string'`

flavianh commented:
Maybe use six instead to handle unicode, since we're on python 2 and 3
Le 1 avr. 2016 08:05, "Cyprien G" <notifications@github.com> a écrit :

> I think you should convert string in conftest.py to unicode: u'string'
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/Trax-air/swagger-parser/pull/11#issuecomment-204263777>
>

TNGPS commented:
finally all lights green !

cyprieng commented:
Your modifications are now available in the last release on pypi. Thanks for the contribution !
